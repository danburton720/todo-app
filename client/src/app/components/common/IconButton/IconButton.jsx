import React from 'react';

export default class IconButton extends React.PureComponent {
    render() {
        const Icon = this.props.icon;
        return (
            <button onClick={this.props.onClick}>
                <Icon size={this.props.size}/>
            </button>
        );
    }
}