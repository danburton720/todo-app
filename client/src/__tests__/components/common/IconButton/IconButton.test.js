import React from 'react';
import {shallow} from 'enzyme';
import IconButton from '../../../../app/components/common/IconButton/IconButton';

describe('IconButton', () => {
    let shallowWrapper;
    let props;
    const wrapper = () => {
        if (!shallowWrapper) {
            shallowWrapper = shallow(<IconButton {...props}/>);
        }
        return shallowWrapper;
    };

    beforeEach(() => {
        props = {
            onClick: jest.fn(),
            size: 20,
            icon: () => <div/>
        };
        shallowWrapper = undefined;
    });

    describe('render', () => {
        it('should return a button with an Icon component', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        describe('button', () => {
            describe('onClick', () => {
                it('should call the prop onClick', () => {
                    expect(props.onClick).not.toHaveBeenCalled();
                    wrapper().find('button').first().props().onClick();
                    expect(props.onClick).toHaveBeenCalledTimes(1);
                });
            });
        });
    });
});