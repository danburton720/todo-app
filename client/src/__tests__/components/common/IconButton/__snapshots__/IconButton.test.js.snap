// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton render should return a button with an Icon component 1`] = `
<button
  className="iconButton"
  onClick={[MockFunction]}
>
  <icon
    size={20}
  />
</button>
`;
